humblebundle_downloader/cli.py

```diff
@@ -13,13 +14,47 @@
 # Ignore unwanted logs from the requests lib when debuging
 logging.getLogger('urllib3.connectionpool').setLevel(logging.WARNING)
 
+# convert a string representing size to an integer 
+def parse_size(size):
+    # Base10 unit definitions  
+    # K=x1000  M=x10000000  G=x1000000000
+    # units = {"K": 10**3, "M": 10**6, "G": 10**9, "T": 10**12}
+    # Binary unit definitions:
+    # K=x1024  M=x1024x1024 etc
+    units = {"K": 2**10, "M": 2**20, "G": 2**30, "T": 2**40}
+    try:
+        return int(size)    # if it's an int already just return it
+    except ValueError:  # it wasn't an int
+        size=size.upper()
+        if size.endswith('B'):
+            return parse_size(size[:-1])
+        unit=size[-1:]
+        number=size[:-1]
+        if unit not in units.keys():
+            raise ValueError(f'Invalid Unit: {unit}')
+        return int(float(number)*units[unit])
+
+# convert a string represting time to an integer number of seconds
+def parse_seconds(size):
+    # convert parameter to number of seconds
+    units = {"S": 1, "M": 60, "H": 60*60, "D": 60*60*24, 'W': 60*60*24*7}
+    try:
+        return int(size)    # if it's an int already just return it
+    except ValueError:  # it wasn't an int
+        size=size.upper()
+        unit=size[-1:]
+        number=size[:-1]
+        if unit not in units.keys():
+            raise ValueError(f'Invalid Unit: {unit}')
+        return int(float(number)*units[unit])
```

Comment:
https://pypi.org/project/humanfriendly/ is a library option for this functionality